### PR TITLE
Raw container support

### PR DESCRIPTION
**Pull request by kumare3**
ContainerTask implemented and works!

### CODE REVIEW

File: flytekit/annotated/task.py [hunk 1]
```diff
@@ -472,7 +593,7 @@ def execute(self, **kwargs) -> Any:
 
 class DynamicWorkflowTask(PythonFunctionTask):
     def __init__(self, dynamic_workflow_function: Callable, metadata: _task_model.TaskMetadata, *args, **kwargs):
-        super().__init__(dynamic_workflow_function, metadata, *args, **kwargs)
+        super().__init__(dynamic_workflow_function, metadata, task_type="dynamic_task", *args, **kwargs)
```
Comment (wild-endeavor) [inline flytekit/annotated/task.py:596]: Can we copy in the current strings in `SdkTaskType` so that the rest of the system respects it as well?

**Review by wild-endeavor — COMMENTED** [review]
did you want to do a mocking example in this PR?

File: flytekit/annotated/task.py (hunk 1 above)
Comment (kumare3) [reply flytekit/annotated/task.py:596]: sure will do

**Comment by kumare3** [comment]
I will do the mocking as a separate PR.

**Pull request merged**